cmd/storage-rpc-client.go

```diff
@@ -93,15 +93,14 @@ func toStorageErr(err error) error {
 }
 
 // Initialize new rpc client.
-func newRPCClient(ep storageEndPoint) (StorageAPI, error) {
-	// Input validation.
-	if ep.host == "" || ep.port == 0 || ep.path == "" {
+func newPosixRPC(ep *url.URL) (StorageAPI, error) {
```

Comment:
newRPCClient is too generic its closer to 'net/rpc' .Also here RPC client is a StorageAPI. 

newRemoteStorage() makes sense.